db/migrate/20240903024311_backfill_user_passwords_from_users.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+class BackfillUserPasswordsFromUsers < ActiveRecord::Migration[7.1]
+  disable_ddl_transaction!
+  BATCH_SIZE = 50_000
+
+  def up
+    min_id, max_id = DB.query_single(<<~SQL.squish)
+      SELECT MIN(id), MAX(id)
+      FROM users
+      WHERE password_hash IS NOT NULL;
```

Comment:
```suggestion
      WHERE password_hash IS NOT NULL AND salt IS NOT NULL AND password_algorithm IS NOT NULL;
```

I think we need to match the condition below?